### ISSUE TITLE
KARAF_ETC is not valid: /openhab/userdata/etc

### ISSUE THREAD

**Issue by fex01** [openhab/openhab-docker#275]
Hi guys,

It seems like the container doesn't start successfully if you are either missing precreated folders (conf, userdata & addons) or if have already content in these folders.

I did also write a post in the openHAB forum: https://community.openhab.org/t/docker-restarting-karaf-etc-is-not-valid-openhab-userdata-etc

The short version:

My docker compose section for openHAB:
```
openhab:
    image: "openhab/openhab:2.5.0"
    container_name: openhab
    restart: always
    network_mode: host
    volumes:
      - "/etc/localtime:/etc/localtime:ro"
      - "/etc/timezone:/etc/timezone:ro"
      - "${PWD}/openhab/addons:/openhab/addons"
      - "${PWD}/openhab/conf:/openhab/conf"
      - "${PWD}/openhab/userdata:/openhab/userdata"
    environment:
      OPENHAB_HTTP_PORT: "8080"
      OPENHAB_HTTPS_PORT: "8443"
      USER_ID: ${UID}
      GROUP_ID: ${GID}
      EXTRA_JAVA_OPTS: "-Duser.timezone=Europe/Berlin"
    ports:
      - "8007:5007"
      - "8080:8080"
      - "8443:8443"
```

My tests:
1. ~/test1: no precreated folders, no chown / chmod, u/gid set to 1000 (main user) :x: 
2. ~/test2: no precreated folders, chown -R 1000:1000, u/gid set to 1000 :x:
3. ~/test3: no precreated folders, chown -R 9001:9001, u/gid set to 1000 (9001 being the default uid if no uid is set for the openHAB image) :x:
  b. stop - create userdata/etc/ - start again :x:
4. ~/test4: no precreated folders, chown -R 9001:9001, u/gid set to 9001 :x:
  b. stop - create userdata/etc/ - start again :x:
5. ~/test5: no precreated folders, chmod -R 777, u/gid set to 1000 :x:
  b. remove container - create userdata/etc/ - start again :x:
6. ~/test6: no precreated folders, chmod -R 777, chown -R 9001:9001, u/gid set to 9001 :x:
  b. remove container - create userdata/etc/ - start again :x:
7. /opt/openhab7: create user openhab7, precreated folders, chown -R openhab7:openhab7, u/gid set to openhab7's :white_check_mark:
8. ~/test8: precreated folders (./conf, ...), chown -R openhab7:openhab7, u/gid set to openhab7's :white_check_mark:
9. ~/test9: precreated folders in subfolder (./openhab/conf, ./openhab/addons, ...), chown -R openhab7:openhab7, u/gid set to openhab7's :white_check_mark:
10. ~/test10: precreated folders in subfolder, no chown / chmod, u/gid set to 1000 :white_check_mark:

**Comment by wertzui** [comment]
I just started with openHAB and Docker and ran into the same problem.
As soon as there is anything inside the userdata/ folder, the error is shown and the container is restarted.

For me it was the uuid and the openhabcloud/secret files that I put there to get a connection with myopenhab.org.

I think, it would be best if someone could put any files inside any folder before the container is started for the first time to achieve the best portability experience. Then someone could just clone a git repo with a docker-compse.yml and some predefined config values in those mounted folders and start everything using `docker-compose up` without having to do anything manually.

**Comment by wborn** [comment]
By default Docker doesn't initialize volumes if there is any data in it to prevent issues such as files being overwritten or data inconsistencies. The openHAB Docker image adheres to the same principles. See also the discussion in https://github.com/openhab/openhab-docker/issues/249. 

We could update the docs for this or perhaps detect such issues and print some helpful messages.

**Comment by fex01** [comment]
I think an updated doc might be helpful - especially because users new to docker will not be aware, that the openhab folders should be empty on first start.

But as you can see in my test cases, the container does also runs into the same error when there a no precreated folders - is that also default behavior?

It can be very frustrating when you run always into the same error without any helpful message about the cause and how to prevent this error...

**Comment by bwosborne2** [comment]
If you did not pre-create the folders you did not read the installation docs. How would adding a warning about data in folders help those who do not read the documentation?

**Comment by fex01** [comment]
I did quite a lot of different things to test what works and what doesn't. And yes, I did read the docs.
But even if I do everything as described in the docs I sometimes run into said non helpful error message.

Just yesterday I did everything as described in my tc 10 - and this time it didn't work any longer. The container created two subfolders under userdata - but was not able to create the userdata/etc subfolder. 
After stopping the container, deleting the precreated folder and subfolders and restarting the container all necessary folders where generated and the openHAB runtime was able to start...
